tool/merkle.go

```diff
@@ -0,0 +1,60 @@
+/*
+Ported from https://github.com/gohornet/hornet codebase.
```

Comment:
I'm not sure if these two functions should be part of the tool package, or the merkle package....

IMHO it fits more in the merkle package, but maybe @luca-moser should decide that.

Reply:
I agree, this two functions should reside in the merkle package.

Reply:
Actually, it was me who made the change request to move them outside the merkle package :stuck_out_tongue: 

We have `merkleTree.WriteTo` which takes an `io.Writer` and this is the key library function offering standard go support to use it together with files, mem, network...
This function is just a trivial wrapper around `WriteTo` with a file as a writer. Also, there are use-case specific options for this (like filemode, buffered vs non-buffered, compression or not, keeping file open...), which cannot be changed. That's why I don't see this is library functionality.

But its absolutely @luca-moser's call, I just wanted to explain my motivation.

Reply:
You're right. The reason why I wanted this to be moved is that I find it odd to have `tools.LoadMerkleTreeFile` because `tools` is a very generic package name. @karimodm you can leave the two functions in place.

Reply:
That is actually a good point. The Merkle stuff is actually the first and only "tool" we have beside the iota.go lib. So at some point we should think about how to deal with this. Do we want to have this in a new repo, in a new folder, in a new package...